### PR TITLE
Get column names from artis code if artis directory exists

### PR DESCRIPTION
**Pull request by ccollins22**

### CODE REVIEW

File: artistools/packets/__init__.py [hunk 1]
```diff
@@ -64,6 +28,77 @@
 type_ids = dict((v, k) for k, v in types.items())
 
 
+@at.diskcache(savezipped=True)
```
Comment (lukeshingles) [inline artistools/packets/__init__.py:31]: I'm guessing that this function is not so slow that it's faster to read a zipped file from disk every time. I suggest @lru_cache(maxsize=16) instead.

File: artistools/packets/__init__.py [hunk 2]
```diff
@@ -64,6 +28,77 @@
...
+def get_column_names(modelpath):
+    modelpath = Path(modelpath)
+    if Path('artis').is_dir():
```
Comment (lukeshingles) [inline artistools/packets/__init__.py:34]: I think this should be Path(modelpath, 'artis').is_dir()

File: artistools/packets/__init__.py [hunk 3]
```diff
@@ -116,7 +151,8 @@ def emtrue_timestep(packet):
     return dfpackets
 
 
-def readfile_text(packetsfile):
+def readfile_text(packetsfile, modelpath=Path('.')):
+    columns = get_column_names(modelpath)
     try:
         inputcolumncount = len(pd.read_csv(packetsfile, nrows=1, delim_whitespace=True, header=None).columns)
```
Comment (lukeshingles) [inline artistools/packets/__init__.py:157]: If you've read the column names from the artis source code, then the number of columns in the packets*.out should strictly match the number of inferred columns. Currently, the code allows the number of named columns to be longer than the actual columns in the file (for backwards compatibility).

Comment (ccollins22) [reply artistools/packets/__init__.py:157]: Do you want a check added for that?

Comment (lukeshingles) [reply artistools/packets/__init__.py:157]: Yes. Easiest way would be to have get_column_names() return columns, from_sourcecode

If from_sourcecode:
    assert len(columns) == inputcolumncount

**Comment by ccollins22** [comment]
Looks fine to me. Go ahead and merge it

**Pull request merged**